src/transformers/custom-elements-manifest/get-declarations.ts

```diff
@@ -0,0 +1,325 @@
+
+		yield {
+			kind: "method",
+			name: method.name,
+			privacy: method.visibility,
+			description: method.jsDoc?.description,
+			parameters,
+			return: {
+				description: returnDescription,
+				type: typeToSchemaType(context, returnTypeHint || returnType)
+			},
+			inheritedFrom: getInheritedFromReference(declaration, method, context),
+			summary: getSummaryFromJsDoc(method.jsDoc)
+			// TODO: "static"
+		};
+	}
+}
+
+/**
+ * Returns class member docs for a declaration
+ * @param declaration
+ * @param context
+ */
+function getClassMembersForDeclaration(declaration: ComponentDeclaration, context: TransformerContext): schema.ClassMember[] {
+	return [...getClassFieldsForDeclaration(declaration, context), ...getMethodsForDeclaration(declaration, context)];
+}
+
+function* getEventsFromDeclaration(declaration: ComponentDeclaration, context: TransformerContext): IterableIterator<schema.Event> {
+	const visibility = context.config.visibility ?? "public";
+	for (const event of declaration.events) {
+		if (event.visibility === visibility) {
+			const type = event.type?.() ?? { kind: "ANY" };
+			const simpleType = isSimpleType(type) ? type : toSimpleType(type, context.checker);
+			const typeName = simpleType.kind === "GENERIC_ARGUMENTS" ? simpleType.target.name : simpleType.name;
+			yield {
+				description: event.jsDoc?.description,
+				name: event.name,
+				inheritedFrom: getInheritedFromReference(declaration, event, context),
+				type: typeName === null || typeName === undefined || simpleType.kind === "ANY" ? { text: "Event" } : { text: typeName }
+			};
+		}
+	}
+}
+
+function* getSlotsFromDeclaration(declaration: ComponentDeclaration, context: TransformerContext): IterableIterator<schema.Slot> {
+	for (const slot of declaration.slots) {
+		yield {
+			description: slot.jsDoc?.description,
+			name: slot.name ?? ""
+		};
+	}
+}
+
+function* getAttributesFromDeclaration(declaration: ComponentDeclaration, context: TransformerContext): IterableIterator<schema.Attribute> {
+	const visibility = context.config.visibility ?? "public";
+	for (const member of declaration.members) {
+		if (member.visibility === visibility && member.attrName) {
+			const type = getTypeHintFromType(member.typeHint ?? member.type?.(), context.checker, context.config);
+			yield {
+				name: member.attrName,
+				fieldName: member.propName,
+				default: member.default != null ? JSON.stringify(member.default) : undefined,
+				description: member.jsDoc?.description,
+				type: type === undefined ? undefined : { text: type },
+				inheritedFrom: getInheritedFromReference(declaration, member, context)
+			};
+		}
+	}
+}
+
+function* getCSSPropertiesFromDeclaration(
+	declaration: ComponentDeclaration,
+	context: TransformerContext
+): IterableIterator<schema.CssCustomProperty> {
+	for (const cssProperty of declaration.cssProperties) {
+		// TODO (43081j): somehow populate the syntax property
```

Comment:
doubt we have logic to do this already. the old implementation passed a typescript type here, but of course css props don't have those. i suppose we'd need new logic to determine the right syntax, if thats even possible?